clash-lib/src/Clash/Primitives/Xilinx/ClockGen.hs

```diff
@@ -0,0 +1,139 @@
+{-|
+  Copyright   :  (C) 2023, QBayLogic B.V.
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+
+  Blackbox template functions for
+  Clash.Xilinx.ClockGen.{clockWizard,clockWizardDifferential}
+-}
+
+{-# LANGUAGE QuasiQuotes #-}
+
+module Clash.Primitives.Xilinx.ClockGen where
+
+import Clash.Backend (Backend)
+import Clash.Netlist.BlackBox.Util (exprToString)
+import Clash.Netlist.Types
+import Clash.Netlist.Util (stripVoid)
+import Clash.Signal (periodToHz)
+
+import Control.Monad.State (State)
+import qualified Data.String.Interpolate as I
+import Data.Text.Prettyprint.Doc.Extra (Doc)
```

Comment:
https://clash-lang.readthedocs.io/en/latest/hacking-on-clash/style.html#imports :

```suggestion
import Clash.Backend (Backend)
import Clash.Netlist.BlackBox.Util (exprToString)
import Clash.Netlist.Types
import Clash.Netlist.Util (stripVoid)
import Clash.Signal (periodToHz)

import Control.Monad.State (State)
import qualified Data.String.Interpolate as I
import Data.Text.Prettyprint.Doc.Extra (Doc)
```
(`Clash.Signal` is a _related third party import_)

I am aware that our code is not really an example of proper application of these guidelines. Do as I say, not as I do ;-).